### PR TITLE
feat: add generation of type description to schema (#708)

### PR DESCRIPTION
**Pull request by KuboOrk**
**Describe the PR**
Added "description" field generation for swagger scheme from struct documentation

**Relation issue**
https://github.com/swaggo/swag/issues/708

**Additional context**
None

### CODE REVIEW

**Comment by ubogdan** [comment]
@KuboOrk, would you mind adding a test case so code coverage will get back to positive?

File: parser.go [hunk 1]
```diff
@@ -978,6 +980,57 @@ func fullTypeName(pkgName, typeName string) string {
...
+			commentText := strings.TrimSpace(strings.TrimLeft(comment.Text, "/"))
+			attribute := strings.Split(commentText, " ")[0]
+			if strings.ToLower(attribute) != descriptionAttr && !isHandlingDescription {
+				continue
+			}
+
+			if strings.ToLower(attribute) != descriptionAttr && isHandlingDescription {
+				break
+			}
+
```
Comment (ubogdan) [inline parser.go:1025]: For better readiness please rewrite the 2 statements above to:
```go
if strings.ToLower(attribute) != descriptionAttr {
       if isHandlingDescription {
          break
       }
       continue;
}
```

File: README.md [hunk 2]
```diff
@@ -613,13 +613,34 @@ type Account struct {
 ### Description of struct
 
 ```go
+// Account model info
+// @Description User account information
+// @Description with user id and username
 type Account struct {
```
Comment (ubogdan) [inline README.md:619]: I believe you should document the contextual is ignored 
```go
// This comment line will be ignored
```

File: parser.go [hunk 3]
```diff
@@ -978,6 +980,57 @@ func fullTypeName(pkgName, typeName string) string {
 	return typeName
 }
 
+// fillDefinition additionally fills fields in definition (spec.Schema)
+// TODO: If .go file contains many types, it may work for a long time
+func fillDefinition(definition *spec.Schema, file *ast.File, typeSpecDef *TypeSpecDef) {
+	if definition.Description == "" {
```
Comment (ubogdan) [inline parser.go:986]: Please move the if statement at line 957 , so the method will be called only when is necesary.
```go
if definition.Description == "" {
	fillDefinition(definition, typeSpecDef.File, typeSpecDef)
}
```

File: parser.go [hunk 4]
```diff
@@ -978,6 +980,57 @@ func fullTypeName(pkgName, typeName string) string {
...
+					extractDeclarationDescription(typeSpec.Doc, typeSpec.Comment, generalDeclaration.Doc)
+			}
+		}
+	}
+}
+
+// extractDeclarationDescription gets first description
+// from attribute descriptionAttr in commentGroups (ast.CommentGroup)
+func extractDeclarationDescription(commentGroups ...*ast.CommentGroup) (description string) {
+	for _, commentGroup := range commentGroups {
```
Comment (ubogdan) [inline parser.go:1009]: Please refactor to:
```
func extractDeclarationDescription(commentGroups ...*ast.CommentGroup) string {
      var description string
```

**Review by ubogdan — CHANGES_REQUESTED** [review]
Great job till now. Please see my comments regarding code styling.

File: README.md (hunk 2 above)
Comment (KuboOrk) [reply README.md:619]: Do you say about `// Account model info`?

Comment (ubogdan) [reply README.md:619]: Yes.

**Review by ubogdan — APPROVED** [review]
LGTM

**Comment by ubogdan** [comment]
@KuboOrk Thanks for your contribution!

**Pull request merged**